src/backend/distributed/connection/remote_commands.c

```diff
@@ -837,9 +838,9 @@ WaitForAllConnections(List *connectionList, bool raiseInterrupts)
 					if (InterruptHoldoffCount > 0 && (QueryCancelPending ||
 													  ProcDiePending))
 					{
-						/* return immediately in case of cancellation */
-						FreeWaitEventSet(waitEventSet);
-						return;
+						/* break out of event loop immediately in case of cancellation */
```

Comment:
I think it'd be great if we add a comment on why we're not returning here immediately.